cmd/clusterctl/client/cluster/installer.go

```diff
@@ -90,6 +109,58 @@ func installComponentsAndUpdateInventory(components repository.Components, provi
 	return providerInventory.Create(inventoryObject)
 }
 
+// waitForReady waits till the installed components are ready.
+func (i *providerInstaller) waitForReady() error {
+	log := logf.Log
+	log.Info("Waiting for providers to be available...")
+
+	if err := i.waitDeploymentsReady(); err != nil {
+		return err
+	}
+
+	// check for services too?
+
+	return nil
+}
+
+// waitDeploymentsReady waits till the installed deployments are ready.
+func (i *providerInstaller) waitDeploymentsReady() error {
+	for _, components := range i.installQueue {
+		for _, obj := range components.Objs() {
+			if obj.GroupVersionKind().Kind == "Deployment" {
+				if err := i.waitDeploymentReady(obj); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (i *providerInstaller) waitDeploymentReady(deployment unstructured.Unstructured) error {
+	err := wait.Poll(500*time.Millisecond, 30*time.Second, func() (bool, error) {
```

Comment:
Wondering if we should make the timeout configurable (we saw similar request for other timeouts from users working on slow environments)

Reply:
+1